src/app/docs/angular-tree/angular-tree-docs.component.ts

```diff
@@ -1,9 +1,194 @@
 import {
-  Component
+  Component,
+  ViewChild
 } from '@angular/core';
 
+import {
+  SkyDocsDemoControlPanelChange,
+  SkyDocsDemoControlPanelRadioChoice
+} from '@skyux/docs-tools';
+
+import {
+  TreeModel
+} from 'angular-tree-component';
+
+import {
+  ITreeOptions
+} from 'angular-tree-component/dist/defs/api';
+
 @Component({
   selector: 'app-angular-tree-docs',
   templateUrl: './angular-tree-docs.component.html'
 })
-export class AngularTreeDocsComponent { }
+export class AngularTreeDocsComponent {
+
+  public set enableCascading(value: boolean) {
+    this.resetSelection();
+    this.treeOptions.useTriState = value;
+    this.demoSettings.enableCascading = value;
+
+    if (value) {
+      this.demoSettings.selectLeafNodesOnly = false;
+    }
+  }
+
+  public set selectLeafNodesOnly(value: boolean) {
+    this.resetSelection();
+    this.demoSettings.selectLeafNodesOnly = value;
+
+    if (value) {
+      this.demoSettings.enableCascading = false;
+    }
+  }
+
+  public basicOptions: ITreeOptions = {
+    animateExpand: true
+  };
+
+  public demoSettings: any = {};
+
+  public dropdownItems: any = [
+    { name: 'Insert an item at this level', disabled: false },
+    { name: 'Insert an item under this level', disabled: false },
+    { name: 'Move up', disabled: false },
+    { name: 'Move down', disabled: false },
+    { name: 'Move left', disabled: false },
+    { name: 'Move right', disabled: false },
+    { name: 'Edit', disabled: false },
+    { name: 'Delete', disabled: false }
+  ];
+
+  public nodes: any[] = [
+    {
+      id: 1,
+      name: 'Animals',
+      isExpanded: true,
+      children: [
+        {
+          id: 2,
+          name: 'Cats',
+          isExpanded:
+          true, children: [
+            { id: 3, name: 'Burmese' },
+            { id: 4, name: 'Persian' },
+            { id: 5, name: 'Tabby' }
+          ]
+        },
+        {
+          id: 6,
+          name: 'Dogs',
+          isExpanded: true,
+          children: [
+            { id: 7, name: 'Beagle' },
+            { id: 8, name: 'German shepard' },
+            { id: 9, name: 'Labrador retriever' }
+          ]
+        }
+      ]
+    }
+  ];
+
+  public optionsCascading: ITreeOptions = {
+    animateExpand: true,
+    useCheckbox: true,
+    useTriState: true
+  };
+
+  public optionsNoncascading: ITreeOptions = {
+    animateExpand: true,
+    useCheckbox: true,
+    useTriState: false
+  };
+
+  public selectModeChoices: SkyDocsDemoControlPanelRadioChoice[] = [
+    { value: 'multi', label: 'Multi-select' },
+    { value: 'single', label: 'Single-select' }
+  ];
+
+  public treeModeChoices: SkyDocsDemoControlPanelRadioChoice[] = [
+    { value: 'navigation', label: 'Navigation' },
```

Comment:
What is the difference between Navigation and Read-only? Based on the behavior in the demo, the only difference I see is how items are highlighted when selected. In Navigation mode, the label gets a border with rounded corners, plus the item gets a gray screen and a blue bar on the left. In Read-only mode, the label only gets the border with the rounded corners. I'm not sure what functional difference we are trying to indicate here, so just making sure this is what we intend the demo to look like.

Reply:
Navigation highlights nodes when clicked while read-only does not. If that's not the case, there may be something wrong.

Reply:
Nope. That's how it's working. I just couldn't remember. Still not sure what the point of the distinction is, but that's fine. I compared back to the existing demo, and this correctly mimics that one.